confidant/services/keymanager.py

```diff
@@ -10,35 +10,47 @@
 from confidant.lib import cryptolib
 
 logger = logging.getLogger(__name__)
-config = botocore.config.Config(
-    connect_timeout=settings.KMS_CONNECTION_TIMEOUT,
-    read_timeout=settings.KMS_READ_TIMEOUT,
-    max_pool_connections=settings.KMS_MAX_POOL_CONNECTIONS
-)
-auth_kms_client = confidant.clients.get_boto_client(
-    'kms',
-    config={'name': 'keymanager', 'config': config}
-)
-at_rest_kms_client = confidant.clients.get_boto_client(
-    'kms',
-    config={'name': 'keymanager', 'config': config}
-)
-iam_resource = confidant.clients.get_boto_resource('iam')
-
-DATAKEYS = {}
-KEY_METADATA = {}
+
+_DATAKEYS = {}
+_KEY_METADATA = {}
+
+
+def _get_boto_config():
+    return botocore.config.Config(
+        connect_timeout=settings.KMS_CONNECTION_TIMEOUT,
+        read_timeout=settings.KMS_READ_TIMEOUT,
+        max_pool_connections=settings.KMS_MAX_POOL_CONNECTIONS,
+    )
+
+
+def _get_auth_kms_client():
+    return confidant.clients.get_boto_client(
+        'kms',
+        config={'name': 'keymanager', 'config': _get_boto_config()},
+        endpoint_url=settings.KMS_URL,
+    )
+
+
+def _get_at_rest_kms_client():
```

Comment:
This looks to duplicate `_get_auth_kms_client()`, do we need need an additional variable in `settings.py`, where one is for auth and the other is for the at rest client?

Reply:
Oh, interestingly enough, back in [this change](https://github.com/lyft/confidant/commit/d7ccb71d4b1e7d02f8c5cb9e691bf7807ecb1b10#diff-929092871d7658f82a1785ab913e6e47R16-R28) I meant to split these into two clients, but didn't give them separate names, so they actually reference the same client. Let me update this to split them into two clients (so they have their own connection pools).